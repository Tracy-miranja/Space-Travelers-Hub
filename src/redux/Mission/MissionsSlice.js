import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  Mission: [],
};

const missionsSlice = createSlice({
  name: 'Mission',
  initialState,
  reducer: {

  },
});

export default missionsSlice.reducer;
