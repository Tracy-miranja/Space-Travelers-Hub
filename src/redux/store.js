import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './Mission/MissionsSlice';
import rocketReducer from './Rocket/RocketsSlice';

export default configureStore({
  reducer: {
    mission: missionReducer,
    rockets: rocketReducer,
  },
});
