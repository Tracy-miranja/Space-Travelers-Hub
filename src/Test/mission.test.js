import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/store';
import Mission from '../components/mission';

it('renders Mission name and decription ', () => {
  const tree = renderer
    .create(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <Mission />
          </Router>
        </Provider>
      </React.StrictMode>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
