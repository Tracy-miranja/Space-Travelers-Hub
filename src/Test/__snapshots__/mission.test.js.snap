// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Mission correctly 1`] = `
<section
  className="table"
>
  <table
    className="table-bordered"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</section>
`;
